### PR TITLE
Hist2d

### PR DESCRIPTION
**Pull request by cyrraz**
Allow for variable binning and basic function to plot 2D histograms

### CODE REVIEW

File: plothist/plotters.py [hunk 1]
```diff
@@ -146,28 +168,28 @@ def make_hist(data, bins=50, range=[None, None], weights=1):
...
         warnings.warn(
-            f"Only {100*range_coverage:.2f}% of data contained in the binning range ({binning[1]}, {binning[2]})."
+            f"Only {100*range_coverage:.2f}% of data contained in the binning range ({axis.edges[0]}, {axis.edges[-1]})."
         )
 
     return h
 
 
-def make_2d_hist(data, binning, weights=1):
+def make_2d_hist(data, bins=[10, 10], range=(None, None), weights=1):
```
Comment (0ctagon) [inline plothist/plotters.py:192]: type(bins) = tuple maybe?

**Comment by 0ctagon** [comment]
change documentation

File: plothist/plotters.py [hunk 2]
```diff
@@ -238,6 +266,14 @@ def plot_hist(hist, ax, **kwargs):
         )
 
 
+def plot_2d_hist(hist, fig, ax, pcolormesh_kwargs={}, colorbar_kwargs={}):
+    im = ax.pcolormesh(*hist.axes.edges.T, hist.values().T, **pcolormesh_kwargs)
+    divider = make_axes_locatable(ax)
+    cax = divider.append_axes("right", size="5%", pad=0.05)
+    ax.tick_params(axis="x", which="both", top=False, bottom=False)
+    fig.colorbar(im, cax=cax, **colorbar_kwargs)
```
Comment (0ctagon) [inline plothist/plotters.py]: # Get the figure object from the axes
fig_object = ax.figure

File: plothist/plotters.py (hunk 1 above)
Comment (0ctagon) [inline plothist/plotters.py:192]: type(bins) = tuple maybe?

File: examples/python/hep_examples.py [hunk 3]
```diff
@@ -81,9 +81,7 @@
 ## 2.1 Define the histograms
 key1 = "variable_1"
 key2 = "variable_2"
-binning1 = [-10, 0, 10]
-binning2 = [-10, -5, 0, 5, 10]
-binning = [binning1, binning2]
+bins = [[-10,0,10],[-10,-5,0,5,10]]
```
Comment (0ctagon) [inline examples/python/hep_examples.py:84]: add small commentary to describe this binning

File: plothist/plotters.py [hunk 4]
```diff
@@ -146,28 +168,28 @@ def make_hist(data, bins=50, range=[None, None], weights=1):
...
+    h = bh.Histogram(axis, storage=storage)
     h.fill(data, weight=weights, threads=0)
 
     # Check what proportion of the data is in the underflow and overflow bins
     range_coverage = h.values().sum() / h.values(flow=True).sum()
     # Issue a warning in more than 1% of the data is outside of the binning range
     if range_coverage < 0.99:
         warnings.warn(
-            f"Only {100*range_coverage:.2f}% of data contained in the binning range ({binning[1]}, {binning[2]})."
+            f"Only {100*range_coverage:.2f}% of data contained in the binning range ({axis.edges[0]}, {axis.edges[-1]})."
```
Comment (0ctagon) [inline plothist/plotters.py:186]: Why [-1] except of [2]?

Comment (cyrraz) [reply plothist/plotters.py:186]: Because this is the list of all bin edges, including internal edges.

**Comment by cyrraz** [comment]
I implemented the comments. Please approve and merge

**Review by 0ctagon — APPROVED** [review]

**Pull request merged**